shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -101,6 +104,24 @@ content::RenderFrame* GetRenderFrame(v8::Local<v8::Value> value) {
   return content::RenderFrame::FromWebFrame(frame);
 }
 
+bool SpellCheckWord(v8::Isolate* isolate,
+                    v8::Local<v8::Value> window,
+                    const std::string& word,
+                    std::vector<base::string16>* optional_suggestions) {
+  size_t start;
+  size_t length;
+
+  ElectronRendererClient* client = ElectronRendererClient::Get();
+  auto* render_frame = GetRenderFrame(window);
+  if (!render_frame)
+    return true;
+
+  base::string16 w = base::UTF8ToUTF16(word);
+  int id = render_frame->GetRoutingID();
+  return client->GetSpellCheck()->SpellCheckWord(
+      w.c_str(), 0, word.size(), id, &start, &length, optional_suggestions);
```

Comment:
Hang on, I think we can just specify `kNoTag` for the tag? Which would mean we wouldn't need to fetch the render frame, which makes me think maybe this API shouldn't be on `webFrame` at all?

Reply:
hm... https://source.chromium.org/chromium/chromium/src/+/master:components/spellcheck/renderer/spellcheck_provider.cc;l=310;drc=c83ae1d48c762639e626be9be0c77a209fdea359 is a reference that uses the routing id. there are other places in the code that use `kNoTag`. cc @MarshallOfSound for further review

Reply:
It seems to be used in the macOS platform spellchecker impl: https://source.chromium.org/chromium/chromium/src/+/master:components/spellcheck/browser/spellcheck_platform_mac.mm;l=201;drc=da08b7363d0947b0e7d98c5282b4cca513f62100